### PR TITLE
Cleaned up how magic modules works document

### PR DESCRIPTION
**Pull request by melinath**
Set up a parallel construction that goes from broad organizational concepts to more detailed ones and links correctly to common code locations.

localhost_1313_magic-modules_get-started_how-magic-modules-works_ (2)

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: docs/content/get-started/how-magic-modules-works.md [hunk 1]
```diff
@@ -18,18 +18,43 @@ There are three types of resources supported by Magic Modules: MMv1, Handwritten
 
 ### MMv1
 
-MMv1 is a Ruby-based code generator that implements Terraform Provider Google (TPG) resources from YAML specification files.
+MMv1 consists of a set of "products", which each contain one or more "resources".
```
Comment (zli82016) [inline docs/content/get-started/how-magic-modules-works.md:21]: `contain` should be `contains`?

File: docs/content/get-started/how-magic-modules-works.md [hunk 2]
```diff
@@ -18,18 +18,43 @@ There are three types of resources supported by Magic Modules: MMv1, Handwritten
...
+
+Handwritten resources and datasources can be grouped by "service", which generally corresponds to the API subdomain the resource or datasource interacts with.
+
+In addition to the core implementation, handwritten resources and datasources will also have documentation, tests, and sweepers (which clean up stray resources left behind by tests). Each type of code is stored in the following locations:
+
+- Resource implementation: In the appropriate service folder inside [`magic-modules/mmv1/third_party/terraform/services`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/services)
+- Resource documentation: [`magic-modules/mmv1/third_party/terraform/website/docs/r`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/r)
+- Datasource implementation: [`magic-modules/mmv1/third_party/terraform/data_sources`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/data_sources)
```
Comment (zli82016) [inline docs/content/get-started/how-magic-modules-works.md:45]: The implementation of the datasources is also moved to the services packages. There is a pending PR to move the rest plugin framework implemented data sources to the service packages.

**Review by zli82016 — COMMENTED** [review]
Thanks for updating the doc. Left comments.

File: docs/content/get-started/how-magic-modules-works.md (hunk 2 above)
Comment (melinath) [reply docs/content/get-started/how-magic-modules-works.md:45]: updated in 227b0b7f0

File: docs/content/get-started/how-magic-modules-works.md (hunk 1 above)
Comment (melinath) [reply docs/content/get-started/how-magic-modules-works.md]: Reworked in ff1a977b4

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**